public/js/index.js

```diff
@@ -524,6 +524,62 @@ Index.loadContextMenuCategories = (catList, id) => Server.callAPI(`/api/archives
     },
 );
 
+/**
+ * Build rating options for contextMenu and select the one for the current ID.
+ * @param {*} id The ID of the archive to check
+ * @returns Ratings
+ */
+Index.loadContextMenuRatings = (id) => Server.callAPI(`/api/archives/${id}/metadata`, "GET", null, `Error finding metadata for ${id}!`,
+    (data) => {
+        const items = {};
+        const ratings = [{
+            id: 1,
+            name: "⭐",
+        }, {
+            id: 2,
+            name: "⭐⭐",
+        }, {
+            id: 3,
+            name: "⭐⭐⭐",
+        }, {
+            id: 4,
+            name: "⭐⭐⭐⭐",
+        }, {
+            id: 5,
+            name: "⭐⭐⭐⭐⭐",
+        }];
+        const tags = data.tags.split(",").map((x) => x.split(":"));
+        const hasRating = tags.some((x) => x[0] === "rating");
+
+        items.push({
+            name: "",
+            events: {
+                click() {
+                    Server.updateTagsFromArchive(id, tags.filter((x) => x[0] !== "rating").map((x) => x.join(":")).join(","));
+                },
+            },
+        });
+
+        for (let i = 0; i < ratings.length; i++) {
+            items[i] = ratings[i];
+            items[i].events = {
+                click() {
+                    if (hasRating) {
+                        const ratingIndex = tags.findIndex((x) => x[0] === "rating");
+                        tags[ratingIndex][1] = ratings[i].name;
+                    } else {
+                        tags.push(["rating", ratings[i]]);
+                    }
+
+                    Server.updateTagsFromArchive(id, tags.map((x) => x.join(":")).join(","));
```

Comment:
Once this goes through, the tags displayed in the main search index don't get updated.  
Not quite sure what'd be the best way to fix this... Maybe just reloading the table would be enough.

Reply:
Fixed on new commit.